cmd/lncli/chainrpc_active.go

```diff
@@ -0,0 +1,200 @@
+//go:build chainrpc
+// +build chainrpc
+
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lnrpc/chainrpc"
+	"github.com/urfave/cli"
+)
+
+// chainCommands will return the set of commands to enable for chainrpc builds.
+func chainCommands() []cli.Command {
+	return []cli.Command{
+		{
+			Name:     "chain",
+			Category: "On-chain",
+			Usage:    "Interact with the bitcoin blockchain.",
+			Subcommands: []cli.Command{
+				getBlockCommand,
+				getBestBlockCommand,
+				getBlockHashCommand,
+			},
+		},
+	}
+}
+
+func getChainClient(ctx *cli.Context) (chainrpc.ChainKitClient, func()) {
+	conn := getClientConn(ctx, false)
+
+	cleanUp := func() {
+		conn.Close()
+	}
+
+	return chainrpc.NewChainKitClient(conn), cleanUp
+}
+
+var getBlockCommand = cli.Command{
+	Name:        "getblock",
+	Category:    "On-chain",
+	Usage:       "Get block by block hash.",
+	Description: "Returns a block given the corresponding block hash.",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "hash",
+			Usage: "the target block hash",
+		},
+		cli.BoolFlag{
+			Name:  "verbose",
+			Usage: "print entire block as JSON",
+		},
+	},
+	Action: actionDecorator(getBlock),
+}
+
+func getBlock(ctx *cli.Context) error {
+	ctxc := getContext()
+
+	var (
+		args            = ctx.Args()
+		blockHashString string
+	)
+
+	verbose := false
+	if ctx.IsSet("verbose") {
+		verbose = true
+	}
+
+	switch {
+	case ctx.IsSet("hash"):
+		blockHashString = ctx.String("hash")
+
+	case args.Present():
+		blockHashString = args.First()
+
+	default:
+		return fmt.Errorf("hash argument missing")
+	}
+
+	blockHash, err := hex.DecodeString(blockHashString)
```

Comment:
We need to use `chainhash.NewHashFromString()` here to do the reversal from human-readable to raw byte representation order.